src/components/plots/DoubletScoresPlot.jsx

```diff
@@ -4,31 +4,91 @@ import PropTypes from 'prop-types';
 import { Vega } from 'react-vega';
 
 import { Skeleton } from 'antd';
+
 import PlatformError from '../PlatformError';
-import generateSpec from '../../utils/plotSpecs/generateDoubletScoresSpec';
+import { generateSpec, generateData } from '../../utils/plotSpecs/generateDoubletScoresSpec';
 import loadCellMeta from '../../redux/actions/cellMeta';
+import { loadCellSets } from '../../redux/actions/cellSets';
+import { loadEmbedding } from '../../redux/actions/embedding';
+import { loadProcessingSettings } from '../../redux/actions/experimentSettings';
 
 const DoubletScoresPlot = (props) => {
   const {
-    experimentId, config, plotData, actions,
+    experimentId, config, actions, plotUuid,
   } = props;
+  const embeddingType = 'umap';
   const dataName = 'doubletScores';
 
   const dispatch = useDispatch();
 
+  const cellSets = useSelector((state) => state.cellSets);
   const doubletScores = useSelector((state) => state.cellMeta?.doubletScores);
+  const embeddingSettings = useSelector(
+    (state) => state.experimentSettings.processing?.configureEmbedding?.embeddingSettings,
+  );
+  const {
+    data: embeddingData,
+    loading: embeddingLoading,
+    error: embeddingError,
+  } = useSelector((state) => state.embeddings[embeddingSettings.method]) || {};
 
   const [plotSpec, setPlotSpec] = useState({});
-  const plotComponent = useSelector((state) => state.componentConfig.embeddingPreviewDoubletScore);
+  const plotComponent = useSelector((state) => state.componentConfig[plotUuid]);
+
+  useEffect(() => {
+    if (doubletScores.loading && !doubletScores.error) {
+      dispatch(loadCellMeta(experimentId, dataName));
+    }
+  }, [experimentId]);
+
+  useEffect(() => {
+    if (cellSets.loading && !cellSets.error) {
+      dispatch(loadCellSets(experimentId));
+    }
+
+    if (!embeddingSettings) {
+      dispatch(loadProcessingSettings(experimentId, embeddingType));
+    }
+
+    if (!embeddingData && embeddingSettings.method) {
+      dispatch(loadEmbedding(experimentId, embeddingSettings.method));
+    }
+  }, [experimentId, embeddingSettings.method]);
```

Comment:
do you know if useEffect gets triggered when all of the items in the `[ ]` brackets change, or is it enough for at least one of them to change for useEffect to get triggered?

Reply:
like in here, will this `useEffect` be triggered only after `experimentId` AND `embeddingSettings.method` get triggered?

Reply:
It'll trigger if any of the values in the dependency array is changed. Reference : https://blog.logrocket.com/guide-to-react-useeffect-hook/